torch/testing/_internal/distributed/distributed_test.py

```diff
@@ -289,6 +289,9 @@ def _run(cls, rank, test_name, file_name):
         self = cls(test_name)
         self.rank = rank
         self.file_name = file_name
+
+        if torch.cuda.device_count() < int(self.world_size):
```

Comment:
Thanks. Is there a reason that we have to compare `device_count` with `self.world_size`? Would 
`if torch.cuda.devide_count() == 1: sys.exit(...)` or `if self.world_size == 1: sys.exit(...)` be enough?

Reply:
The test can be run by manually setting the WORLD_SIZE through an environment variable. In that case, we could have scenarios like: 1) device_count = 8, world_size=4 2) device_count=4, world_size=8. The check ensures we have enough GPUs for the given world_size.

Reply:
Oops, I missed that we can set env var of `WORLD_SIZE` manually for the test. Thanks for pointing it out!